### ISSUE TITLE
Introduce "hubs" of restaurants/shops

### ISSUE THREAD

**Issue by alexsegura** [coopcycle/coopcycle-web#1792]
The goal of this is to fix the problem where multiple restaurants/shops are located at the same address.

For example, a market where there are multiple shops. 
In this case, it is easy to optimize things on the logistics point of view, because there is only one pickup. 
Customers should be able to add products from multiple restaurants/shops to the cart, pay the whole order, and then we can go pick everything at a unique location. 

A "simple" solution for this could be to introduce a concept of "hub", i.e a composite list of restaurants/shops.
From the UI/UX point of view, there would be an additional level: when the customer clicks on the hub, it shows the list of restaurants/shops belonging to the hub. 

#### Fitting with current models/entities

The main problem we have now, is that an order is attached to a restaurant. 
When adding products to the cart, the system checks if the product belongs to the restaurant attached to the cart. 

Obviously, the "hub" is a different class/object, it's not a `LocalBusiness`
So we would need to introduce a new class, but it's quite hard to remove this link between order & restaurant right now. 
Lots of things are depending on it, like access control, Stripe credentials, etc...

TL;DR the problem is we have a `restaurant_id` column in the `sylius_order` table

It should be possible to remove this limitation, while maintaining a "soft" relationship, by introducing an intermediary class, to represent the "target" of the order. The target can point to a single restaurant, or to multiple restaurants. 

#### Splitting the money between multiple restaurants/shops

When ordering at a hub, the customer pays once for the whole order. 
But we still need to give to each restaurant a part of the order, corresponding to its products.
It should be easy to do this using [Stripe Transfers](https://stripe.com/docs/connect/charges-transfers).
It means the Stripe fees are paid by the platform, however, no other choice. 

Also see #1311

**TODO**

- [x] `PreparationTimeCalculator` should iterate over restaurants, and return the highest preparation time

**Comment by agichim** [comment]
One issue I see is that we need to have stock levels for each products. 

If Christmas is crazy busy and the shop owners all they do all day is sell and re-stock their shop, they shouldn't worry about the platform if they run out of stock. 
So, if customers order 50 christmas trees and the shop has 40 christmas trees in stock and they run out but can only replenish stock the next day than 10 customers will have paid but they won't be getting their product. 

So by implementing automatic stock levels, the shop owners just have to put in a number of items for each product when they create their product and they don't have to worry about running out of items. And when they replenish stock they update that number. 

I feel like this is super important, otherwise with 20 small christmas businesses in the Hub without stock level control it can all go out of hand.
